agent/task.go

```diff
@@ -203,6 +210,12 @@ func (tm *taskManager) run(ctx context.Context) {
 			if cancel != nil {
 				// cancel outstanding operation.
 				cancel()
```

Comment:
I did the test with this patch applied and the problem of orphan containers was still there. The race illustrated in #1159 did not get fixed. The point is that the calling to cancel() is disruptive to the ongoing create operation. It causes the create operation return without waiting for the response from docker daemon because of context cancelled. Then the agent task proceeds to remove the container, while the docker daemon is creating it.  When the remove request comes to the daemon before the container was created, the remove operation will fail and the container will become orphan after it's created.

In simple words, the problem could happen if the cancel() is called when the agent task is still creating the container.